x-pack/plugins/elastic_assistant/server/lib/langchain/llm/actions_client_llm.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest } from '@kbn/core/server';
+import type { PluginStartContract as ActionsPluginStart } from '@kbn/actions-plugin/server';
+import { LLM } from 'langchain/llms/base';
+import { get } from 'lodash/fp';
+
+import { getMessageContentAndRole } from '../helpers';
+
+const LLM_TYPE = 'ActionsClientLlm';
+
+export class ActionsClientLlm extends LLM {
+  #actions: ActionsPluginStart;
+  #connectorId: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  #request: KibanaRequest<unknown, unknown, any, any>;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  #actionResultData: Record<string, any>;
+
+  constructor({
+    actions,
+    connectorId,
+    request,
+  }: {
+    actions: ActionsPluginStart;
+    connectorId: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    request: KibanaRequest<unknown, unknown, any, any>;
+  }) {
+    super({});
+
+    this.#actions = actions;
+    this.#connectorId = connectorId;
+    this.#request = request;
+    this.#actionResultData = {};
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  getActionResultData(): Record<string, any> {
+    return this.#actionResultData;
+  }
+
+  _llmType() {
+    return LLM_TYPE;
+  }
+
+  async _call(prompt: string): Promise<string> {
+    // convert the Langchain prompt to an assistant message:
+    const assistantMessage = getMessageContentAndRole(prompt);
+
+    // create a new connector request body with the assistant message:
+    const requestBody = {
+      actionId: this.#connectorId,
+      params: {
+        ...this.#request.body.params, // the original request body params
+        subActionParams: {
+          ...this.#request.body.params.subActionParams, // the original request body params.subActionParams
+          body: JSON.stringify({ messages: [assistantMessage] }),
```

Comment:
The `stringify` here is for the actions framework as it takes a raw string instead of a structure body.